src/main/java/com/auth/utils/JwtUtils.java

```diff
@@ -6,72 +6,70 @@
 import java.util.function.Function;
 
 import org.springframework.security.core.userdetails.UserDetails;
-import org.springframework.stereotype.Component;
 import org.springframework.stereotype.Service;
 
+import com.auth.models.CustomUserDetails;
+
 import io.jsonwebtoken.Claims;
 import io.jsonwebtoken.Jwts;
 import io.jsonwebtoken.SignatureAlgorithm;
 
 @Service
 public class JwtUtils {
 
-
-		private static final long serialVersionUID = -2550185165626007488L;
-
-		public static final long JWT_TOKEN_VALIDITY = 5 * 60 * 60;
-
-		
-		private String secret = "Bharadwaj";
-
-		//retrieve username from jwt token
-		public  String extractUsername(String token) {
-			return extractClaim(token, Claims::getSubject);
-		}
-
-		//retrieve expiration date from jwt token
-		public Date getExpirationDateFromToken(String token) {
-			return extractClaim(token, Claims::getExpiration);
-		}
-
-		public <T> T extractClaim(String token, Function<Claims, T> claimsResolver) {
-			final Claims claims = extractAllClaims(token);
-			return claimsResolver.apply(claims);
-		}
-	    //for retrieveing any information from token we will need the secret key
-		private Claims extractAllClaims(String token) {
-			return Jwts.parser().setSigningKey(secret).parseClaimsJws(token).getBody();
-		}
-
-		//check if the token has expired
-		private Boolean isTokenExpired(String token) {
-			final Date expiration = getExpirationDateFromToken(token);
-			return expiration.before(new Date());
-		}
-
-		//generate token for user
-		public String generateToken(UserDetails userDetails) {
-			Map<String, Object> claims = new HashMap<>();
-			return createToken(claims, userDetails.getUsername());
-		}
-
-		//while creating the token -
-		//1. Define  claims of the token, like Issuer, Expiration, Subject, and the ID
-		//2. Sign the JWT using the HS512 algorithm and secret key.
-		//3. According to JWS Compact Serialization(https://tools.ietf.org/html/draft-ietf-jose-json-web-signature-41#section-3.1)
-		//   compaction of the JWT to a URL-safe string 
-		private String createToken(Map<String, Object> claims, String subject) {
-
-			return Jwts.builder().setClaims(claims).setSubject(subject).setIssuedAt(new Date(System.currentTimeMillis()))
-					.setExpiration(new Date(System.currentTimeMillis() + JWT_TOKEN_VALIDITY * 1000))
-					.signWith(SignatureAlgorithm.HS512, secret).compact();
-		}
-
-		//validate token
-		public Boolean validateToken(String token, UserDetails userDetails) {
-			final String username = extractUsername(token);
-			return (username.equals(userDetails.getUsername()) && !isTokenExpired(token));
-		}
-	
+	public static final long JWT_TOKEN_VALIDITY = 5 * 60 * 60;
```

Comment:
We can have this in the configuration file and read it from there.

Reply:
config file kante we have something called vaults ..daniloki will migrate

Reply:
We will use the vaults while **deploying** the configuration files, reading from the vault for each config during an API call is expensive and not in scope for now. The repo must be having a configuration file, I'll add it soon :)